src/order/hom/lattice.lean

```diff
@@ -0,0 +1,549 @@
+instance : inhabited (lattice_hom α α) := ⟨lattice_hom.id α⟩
+
+@[simp] lemma coe_id : ⇑(lattice_hom.id α) = id := rfl
+
+variables {α}
+
+@[simp] lemma id_apply (a : α) : lattice_hom.id α a = a := rfl
+
+end lattice_hom
+
+@[priority 100] -- See note [lower instance priority]
+instance lattice_hom_class.to_inf_hom_class [lattice α] [lattice β] [lattice_hom_class F α β] :
+  inf_hom_class F α β :=
+{ .. ‹lattice_hom_class F α β› }
+
+namespace bounded_lattice_hom
+variables [lattice α] [lattice β] [bounded_order α] [bounded_order β]
+
+/-- Reinterpret a `bounded_lattice_hom` as a `top_hom`. -/
+def to_top_hom (f : bounded_lattice_hom α β) : top_hom α β := { ..f }
+
+/-- Reinterpret a `bounded_lattice_hom` as a `bot_hom`. -/
+def to_bot_hom (f : bounded_lattice_hom α β) : bot_hom α β := { ..f }
+
+instance : bounded_lattice_hom_class (bounded_lattice_hom α β) α β :=
+{ coe := λ f, f.to_fun,
+  coe_injective' := λ f g h, by obtain ⟨⟨⟨_, _⟩, _⟩, _⟩ := f; obtain ⟨⟨⟨_, _⟩, _⟩, _⟩ := g; congr',
+  map_sup := λ f, f.map_sup',
+  map_inf := λ f, f.map_inf',
+  map_top := λ f, f.map_top',
+  map_bot := λ f, f.map_bot' }
+
+/-- Helper instance for when there's too many metavariables to apply `fun_like.has_coe_to_fun`
+directly. -/
+instance : has_coe_to_fun (bounded_lattice_hom α β) (λ _, α → β) := ⟨λ f, f.to_fun⟩
+
+@[simp] lemma to_fun_eq_coe {f : bounded_lattice_hom α β} : f.to_fun = (f : α → β) := rfl
+
+@[ext] lemma ext {f g : bounded_lattice_hom α β} (h : ∀ a, f a = g a) : f = g := fun_like.ext f g h
+
+/-- Copy of a `bounded_lattice_hom` with a new `to_fun` equal to the old one. Useful to fix
+definitional equalities. -/
+protected def copy (f : bounded_lattice_hom α β) (f' : α → β) (h : f' = f) :
+  bounded_lattice_hom α β :=
+{ to_fun := f',
+  .. f.to_lattice_hom.copy f' $ by { ext, exact congr_fun h _ },
+  .. f.to_top_hom.copy f' $ by { ext, exact congr_fun h _ },
+  .. f.to_bot_hom.copy f' $ by { ext, exact congr_fun h _ } }
+
+variables (α)
+
+/-- `id` as an `bounded_lattice_hom`. -/
+protected def id : bounded_lattice_hom α α :=
+{ to_fun := id,
+  map_sup' := λ _ _, rfl,
+  map_inf' := λ _ _, rfl,
+  map_top' := rfl,
+  map_bot' := rfl }
+
+instance : inhabited (bounded_lattice_hom α α) := ⟨bounded_lattice_hom.id α⟩
+
+@[simp] lemma coe_id : ⇑(bounded_lattice_hom.id α) = id := rfl
+
+variables {α}
+
+@[simp] lemma id_apply (a : α) : bounded_lattice_hom.id α a = a := rfl
+
+end bounded_lattice_hom
+
+@[priority 100] -- See note [lower instance priority]
+instance bounded_lattice_hom_class.to_top_hom_class [lattice α] [lattice β]
+  [bounded_order α] [bounded_order β] [bounded_lattice_hom_class F α β] :
+  top_hom_class F α β :=
+{ .. ‹bounded_lattice_hom_class F α β› }
+
+@[priority 100] -- See note [lower instance priority]
+instance bounded_lattice_hom_class.to_bot_hom_class [lattice α] [lattice β]
+  [bounded_order α] [bounded_order β] [bounded_lattice_hom_class F α β] :
+  bot_hom_class F α β :=
+{ .. ‹bounded_lattice_hom_class F α β› }
```

Comment:
Shouldn't there be instances to `sup_hom_class` and `inf_hom_class` as well?

Reply:
There already is because `bounded_lattice_hom_class extends lattice_hom_class`